build.sbt

```
@@ -82,8 +82,9 @@ lazy val `core-common` = crossProject(JVMPlatform, JSPlatform, NativePlatform)
     name := "otel4s-core-common",
     libraryDependencies ++= Seq(
       "org.typelevel" %%% "cats-core" % CatsVersion,
+      "org.typelevel" %%% "cats-effect" % CatsEffectVersion,
```

kernel dependency should be enough. The downside is that we should update the imports, e.g. `cats.effect.Resource -> cats.effect.kernel.Resource`.

```suggestion
      "org.typelevel" %%% "cats-effect-kernel" % CatsEffectVersion,
```

---

it turns out that core-common didn't even use kernel, but java-common relied on cats-effect via core-common